mypyc/irbuild/specialize.py

```diff
@@ -1046,3 +1057,28 @@ def translate_object_new(builder: IRBuilder, expr: CallExpr, callee: RefExpr) ->
         return builder.add(Call(ir.setup, [subtype], expr.line))
 
     return None
+
+
+@specialize_function("__setattr__", object_rprimitive)
+def translate_object_setattr(builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> Value | None:
+    is_super = isinstance(expr.callee, SuperExpr)
+    is_object_callee = is_object(callee)
+    if not ((is_super and len(expr.args) >= 2) or (is_object_callee and len(expr.args) >= 3)):
+        return None
+
+    self_reg = builder.accept(expr.args[0]) if is_object_callee else builder.self()
+    ir = builder.get_current_class_ir()
```

Comment:
What if this is a non-native class? Do we need to anything different?

Reply:
i think in general we can still translate `object.__setattr__` calls because the underlying implementation of `__setattr__` in cpython just calls the same function that we translate to.

for `super().__setattr__` there might be issues when a non-native class inherits from another non-native class that defines `__setattr__` as the translation would skip that inherited definition. but if it inherits only from object then we should be fine because we're back to `object.__setattr__`. i've changed the conditions for translating `super().__setattr__` to reflect this.